sonic_data_client/db_client.go

```diff
@@ -189,14 +217,49 @@ func (c *DbClient) StreamRun(q *queue.PriorityQueue, stop chan struct{}, w *sync
 		},
 	})
 	log.V(2).Infof("%v Synced", c.pathG2S)
-	for {
-		select {
-		default:
-			time.Sleep(time.Second)
-		case <-c.channel:
-			log.V(1).Infof("Exiting StreamRun routine for Client %v", c.pathG2S)
-			return
+
+	<-c.channel
+	log.V(1).Infof("Exiting StreamRun routine for Client %v", c)
+}
+
+// streamOnChangeSubscription implements Subscription "ON_CHANGE STREAM" mode
+func streamOnChangeSubscription(c *DbClient, gnmiPath *gnmipb.Path) {
+	tblPaths := c.pathG2S[gnmiPath]
+	log.V(2).Infof("streamOnChangeSubscription gnmiPath: %v", gnmiPath)
+
+	if tblPaths[0].field != "" {
+		if len(tblPaths) > 1 {
+			go dbFieldMultiSubscribe(c, gnmiPath, true, time.Millisecond*200, false)
+		} else {
+			go dbFieldSubscribe(c, gnmiPath, true, time.Millisecond*200)
+		}
+	} else {
+		// sample interval and update only parameters are not applicable
```

Comment:
When  is this gets called? When key in table is subscribed instead of single/multiple fields? If yes could you add more elaborative comment lease

Reply:
Yes. Added example queries to the functions documentation for `dbFieldSubscribe`, `dbFieldMultiSubscribe` and `dbTableKeySubscribe`